src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonParameterInfo.cs

```diff
@@ -11,108 +11,43 @@ namespace System.Text.Json.Serialization.Metadata
-                Debug.Assert(_jsonTypeInfo == null);
-                _jsonTypeInfo = value;
-            }
-        }
+        public JsonTypeInfo JsonTypeInfo => MatchingProperty.JsonTypeInfo;
 
-        public Type PropertyType { get; set; } = null!;
+        public Type ParameterType { get; }
 
-        public bool ShouldDeserialize { get; private set; }
+        public bool ShouldDeserialize { get; }
 
-        public JsonPropertyInfo MatchingProperty { get; private set; } = null!;
+        public JsonPropertyInfo MatchingProperty { get; }
 
-        public virtual void Initialize(JsonParameterInfoValues parameterInfo, JsonPropertyInfo matchingProperty, JsonSerializerOptions options)
+        public JsonParameterInfo(JsonParameterInfoValues parameterInfoValues, JsonPropertyInfo matchingProperty)
         {
+            Debug.Assert(matchingProperty.IsConfigured);
+
             MatchingProperty = matchingProperty;
-            ClrInfo = parameterInfo;
-            Options = options;
-            ShouldDeserialize = true;
+            ShouldDeserialize = !matchingProperty.IsIgnored;
+            Options = matchingProperty.Options;
+            Position = parameterInfoValues.Position;
 
-            PropertyType = matchingProperty.PropertyType;
-            NameAsUtf8Bytes = matchingProperty.NameAsUtf8Bytes!;
-            ConverterBase = matchingProperty.EffectiveConverter;
+            ParameterType = matchingProperty.PropertyType;
+            NameAsUtf8Bytes = matchingProperty.NameAsUtf8Bytes;
             IgnoreNullTokensOnRead = matchingProperty.IgnoreNullTokensOnRead;
             NumberHandling = matchingProperty.EffectiveNumberHandling;
-            MatchingPropertyCanBeNull = matchingProperty.PropertyTypeCanBeNull;
-        }
-
-        /// <summary>
-        /// Create a parameter that is ignored at run time. It uses the same type (typeof(sbyte)) to help
-        /// prevent issues with unsupported types and helps ensure we don't accidently (de)serialize it.
-        /// </summary>
-        public static JsonParameterInfo CreateIgnoredParameterPlaceholder(
-            JsonParameterInfoValues parameterInfo,
-            JsonPropertyInfo matchingProperty,
-            bool sourceGenMode)
-        {
-            JsonParameterInfo jsonParameterInfo = new JsonParameterInfo<sbyte>();
-            jsonParameterInfo.ClrInfo = parameterInfo;
-            jsonParameterInfo.PropertyType = matchingProperty.PropertyType;
-            jsonParameterInfo.NameAsUtf8Bytes = matchingProperty.NameAsUtf8Bytes!;
-
-            // TODO: https://github.com/dotnet/runtime/issues/60082.
-            // Default value initialization for params mapping to ignored properties doesn't
-            // account for the default value of optional parameters. This should be fixed.
-
-            if (sourceGenMode)
-            {
-                // The <T> value in the matching JsonPropertyInfo<T> instance matches the parameter type.
```

Comment:
This separation is no longer necessary -- the <T> value now always matches that of the supplied JsonPropertyInfo<T>.